scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3851,431 @@ def rvs(self, dim, size=1, random_state=None):
+    0.25174825174825205
+
+    This broadcasting also works for ``cov``, where the output objects are
+    square matrices of size ``m.shape[-1]``. For example:
+
+    >>> multivariate_hypergeom.cov(m=[[7, 9], [10, 15]], n=[8, 12])
+    array([[[ 1.05, -1.05],
+            [-1.05,  1.05]],
+           [[ 1.56, -1.56],
+            [-1.56,  1.56]]])
+
+    In this example, ``m.shape == (2, 2)`` and ``n.shape == (2,)``,
+    broadcast, following the rules above, as if ``m.shape == (2,)``.
+    Thus the result should also be of shape ``(2,)``, but since each output is
+    a :math:`2 \times 2` matrix. The result, in fact, has shape ``(2, 2, 2)``,
+    where ``result[0]`` is equal to
+    ``multivariate_hypergeom.cov(m=[7, 9], n=8)`` and ``result[1]`` is equal
+    to ``multivariate_hypergeom.cov(m=[10, 15], n=12)``.
+
+    See also
+    --------
+    scipy.stats.hypergeom
+    numpy.random.Generator.multivariate_hypergeometric
+
+    References
+    ----------
+    .. [1] The Multivariate Hypergeometric Distribution,
+           http://www.randomservices.org/random/urn/MultiHypergeometric.html
+
+    .. [2] Thomas J. Sargent and John Stachurski, 2020,
+           Multivariate Hypergeometric Distribution
+           https://python.quantecon.org/_downloads/pdf/multi_hyper.pdf
+    """
+    def __init__(self, seed=None):
+        super(multivariate_hypergeom_gen, self).__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__, mhg_docdict_params)
+
+    def __call__(self, m, n, seed=None):
+        """
+        Create a frozen multivariate_hypergeom distribution.
+
+        See `multivariate_hypergeom_frozen` for more information.
+        """
+        return multivariate_hypergeom_frozen(m, n, seed=seed)
+
+    def _process_parameters(self, m, n):
+        m = np.asarray(m, dtype=np.int_)
+        n = np.asarray(n, dtype=np.int_)
+        if m.ndim == 0:
+            raise ValueError("'m' must be an array!")
+        mcond = np.any(m < 0, axis=-1)
+
+        M = m.sum(axis=-1)
+
+        ncond = (n <= 0) | (n > M)
+        return M, m, n, mcond | ncond
+
+    def _process_quantiles(self, x, M, m, n):
+        xx = np.asarray(x, dtype=np.int_)
+        if xx.ndim == 0:
+            raise ValueError("x must be an array.")
+        if xx.size != 0 and not xx.shape[-1] == m.shape[-1]:
+            raise ValueError("Size of each quantile should be size of 'm': "
+                             "received %d, but expected %d." %
+                             (xx.shape[-1], m.shape[-1]))
+
+        xcond = np.any(xx != x, axis=-1)
+        xcond = xcond | np.any((xx < 0) | (xx > m), axis=-1)
+        xcond = xcond | (xx.sum(axis=-1) != n)
+        return xx, xcond
+
+    def _checkresult(self, result, cond, bad_value):
+        result = np.asarray(result)
+        if cond.ndim != 0:
+            result[cond] = bad_value
+        elif cond:
+            if result.ndim == 0:
+                return bad_value
+            result[...] = bad_value
```

Comment:
![image](https://user-images.githubusercontent.com/6570539/93001794-6f965500-f4e6-11ea-90db-97e1869bbba7.png)

Reply:
Removed. Turns out, the `cond` array will have the same dimensions as the `result` array (due to external broadcasting) and that code checking the condition should be removed.